File: python/sliding-window/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        distance = len(nums) + 1
        left = 0
        subarraySum = 0
        for i, num in enumerate(nums):
            subarraySum += num
            while subarraySum >= target:
                distance = min(distance, i - left + 1)
                subarraySum -= nums[left]
                left += 1

        if distance == len(nums) + 1:
            return 0

        return distance

File: c++/sliding-window/minimum-size-subarray-sum.cc
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int sum = 0;
        int left = 0;
        int length = std::numeric_limits<int>::max();
        
        for (int i = 0; i < nums.size(); ++i) {
            sum += nums[i];
            while (sum >= target) {
                length = std::min(length, i - left + 1);
                sum -= nums[left];
                left++;
            }
        }

        return (length < std::numeric_limits<int>::max()) ? length : 0;
    }
};
